include/openmc/material.h

```diff
@@ -90,6 +90,9 @@ class Material {
   void set_densities(
     const vector<std::string>& name, const vector<double>& density);
 
+  //! Clone the material by deep-copying all members
+  void clone();
```

Comment:
I think we should maybe return a reference to what was just cloned, for convenience. You can still call as if it's a void-returning function.